docs/source/model_design/programming_vivarium/debugging/index.rst

```diff
@@ -47,33 +66,31 @@ Often the immediate line of code that generated the error will be
 significantly removed from the line of code that will need to be changed 
 to fix the error. 
 
-Let's work through an example to understand this. Say we are building an 
-artifact (more information on the :ref:`artifact building page <artifact_building_rt>`) and working 
-on a funciton in the :code:`loader.py` file (more information in the 
-:ref:`overview of engineering files page <engineering_files_rt>`. In this 
-fucntion, a line of code 
-we edit filters a dataset incorrectly 
-such that it is actually empty. Nothing in that immediate block of 
-code catches this error - instead the provided dataframe is just 
-stored as empty. 
-
-From there this same, empty dataset is used in another :code:`loader.py` 
-function to calculate a rate. So our erroreously empty dataframe will be 
-multiplied by a dataframe with the correct data in it. 
-When pandas tries to run this multiplication on the dataframes, 
-it will find that the lengths do not match when they should. Pandas has 
-functions that run through a series of checks and if one fails, throws an 
-error message. 
-
-So the bottom line of the stack trace will likely say an error was thrown 
-in a pandas error message function that the lengths are not equal. From there, 
-the stack trace displays the code that lead here - for example the pandas 
-checking function that was run, then the pandas function that was called 
-(the multiplication), then the secondary 
-:code:`loader.py` function that tried to multiply with the empty dataframe. And still, 
-none of these will actually reveal the issue with filtering the dataset 
-we started with! This can be further complicated if the :code:`loader.py` function 
-uses some functions for other packages, like :code:`vivarium_research` or :code:`vivarium_public_health`. 
+For our example, let's look at two photos of the stack trace 
+output. The first is the immediate output you see. This is quite 
+difficult to read, and as you can see, the bottom lines of the traceback, 
+which are the most immediate code that failed, just show an error in 
+the pandas operating system. Above that, we can start to see the error 
+being traced through calls in the :code:`loader.py` file. 
+
+.. image:: stacktrace_1.png
+
+The second photo is a formatted version of the stack trace that 
+vivarium creates. This is color coded, includes information on the 
```

Comment:
Hmm maybe! I can check with the engineering team. This shows up when I run vivarium based commands, but not otherwise, so I had assumed they might have added some helper functions for this stuff. But lemme ask them and confirm. I like the colors version at any rate - it's easier to read.

Reply:
Yes it's way better! For vivarium commands there is also `--pdb` and if you always add that even when you don't know that you need it, you'll get something even better than a stack trace -- a debugger already open to the point where it failed 😃 Though of course as you mention here sometimes the problem was introduced earlier.